### PR TITLE
Add IOT agent upgrade kitchen tests

### PR DESCRIPTION
**Pull request by kbogtob**
### What does this PR do?

Add kitchen tests for the upgrade of the iot-agent.

### Motivation

Users will have to upgrade the iot-agent soon and we want to test and stabilize this path.

### CODE REVIEW

**Comment by kbogtob** [comment]
Pipeline job: DataDog/datadog-agent/-/jobs/33028898

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1774,6 +1774,12 @@ deploy_windows_testing-a7:
     - .kitchen_datadog_agent_flavor
     - .kitchen_azure_location_north_central_us
 
+.kitchen_test_upgrade_iot_agent:
```
Comment (KSerrania) [inline .gitlab-ci.yml:1777]: Nit: I'd name the jobs with an additional `7`:
```suggestion
.kitchen_test_upgrade7_iot_agent:
```
to match the other jobs' naming convention, and clearly show which configuration (upgrade7) is used in the test.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1979,6 +1985,12 @@ kitchen_centos_upgrade7_agent-a7:
     - .kitchen_scenario_centos_a7
     - .kitchen_test_upgrade7_agent
 
+kitchen_centos_upgrade_iot_agent-a7:
+  allow_failure: false
+  extends:
+    - .kitchen_scenario_centos_a7
+    - .kitchen_test_upgrade_iot_agent
```
Comment (KSerrania) [inline .gitlab-ci.yml:1992]:
```suggestion
kitchen_centos_upgrade7_iot_agent-a7:
  allow_failure: false
  extends:
    - .kitchen_scenario_centos_a7
    - .kitchen_test_upgrade7_iot_agent
```

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -2053,6 +2065,12 @@ kitchen_ubuntu_upgrade7_agent-a7:
     - .kitchen_scenario_ubuntu_a7
     - .kitchen_test_upgrade7_agent
 
+kitchen_ubuntu_upgrade_iot_agent-a7:
+  allow_failure: false
+  extends:
+    - .kitchen_scenario_ubuntu_a7
+    - .kitchen_test_upgrade_iot_agent
```
Comment (KSerrania) [inline .gitlab-ci.yml:2072]:
```suggestion
kitchen_ubuntu_upgrade7_iot_agent-a7:
  allow_failure: false
  extends:
    - .kitchen_scenario_ubuntu_a7
    - .kitchen_test_upgrade7_iot_agent
```

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -2126,6 +2144,12 @@ kitchen_suse_upgrade7_agent-a7:
     - .kitchen_scenario_suse_a7
     - .kitchen_test_upgrade7_agent
 
+kitchen_suse_upgrade_iot_agent-a7:
+  allow_failure: false
+  extends:
+    - .kitchen_scenario_suse_a7
+    - .kitchen_test_upgrade_iot_agent
```
Comment (KSerrania) [inline .gitlab-ci.yml:2151]:
```suggestion
kitchen_suse_upgrade7_iot_agent-a7:
  allow_failure: false
  extends:
    - .kitchen_scenario_suse_a7
    - .kitchen_test_upgrade7_iot_agent
```

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -2194,12 +2218,18 @@ kitchen_debian_upgrade6_agent-a7:
...
   allow_failure: false
   extends:
     - .kitchen_scenario_debian_a7
     - .kitchen_test_upgrade7_agent
 
+kitchen_debian_upgrade_iot_agent-a7:
+  allow_failure: false
+  extends:
+    - .kitchen_scenario_debian_a7
+    - .kitchen_test_upgrade_iot_agent
```
Comment (KSerrania) [inline .gitlab-ci.yml:2231]:
```suggestion
kitchen_debian_upgrade7_iot_agent-a7:
  allow_failure: false
  extends:
    - .kitchen_scenario_debian_a7
    - .kitchen_test_upgrade7_iot_agent
```

**Review by KSerrania — APPROVED** [review]
LGTM

**Pull request merged**